pkg/datapath/linux/ipsec/ipsec_linux_test.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build privileged_tests
+
+package ipsec
+
+import (
+	"net"
+	"os"
+	"testing"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/vishvananda/netlink"
+)
+
+// Hook up gocheck into the "go test" runner.
+func Test(t *testing.T) { TestingT(t) }
+
+type IPSecSuitePrivileged struct{}
+
+var _ = Suite(&IPSecSuitePrivileged{})
+
+var (
+	path             = "ipsec_keys_test"
+	dfltKeysDat    = []byte("hmac(sha256) abcdefghijklmnopqrstuvwzyzABCDEF cbc(aes) abcdefghijklmnopqrstuvwzyzABCDEF\n")
+	scopedKeysDat  = []byte("hmac(sha256) abcdefghijklmnopqrstuvwzyzABCDEF cbc(aes) abcdefghijklmnopqrstuvwzyzABCDEF foobar\n")
+	invalidKeysDat = []byte("test abcdefghijklmnopqrstuvwzyzABCDEF test abcdefghijklmnopqrstuvwzyzABCDEF\n")
+)
+
+func (p *IPSecSuitePrivileged) TestLoadKeysNoFile(c *C) {
+	err := LoadIPSecKeysFile(path)
+	c.Assert(err, ErrorMatches, "failed to load IPSec Keys File ipsec_keys_test: open ipsec_keys_test: no such file or directory")
```

Comment:
nit: Would be nice if this was something like a `c.Assert(os.IsNotExist(err), Equals, true)` just in case the error itself gets reworded at some point.

Reply:
lgtm.

Reply:
Note: This would require use of `os.PathError{}` in the `LoadIPSecKeysFile()`, otherwise the type won't be correct and `os.IsNotExist()` needs to understand the type to work properly.